From sangsu@gmail.com  Mon Jan  5 16:12:00 2009
From: SangSu Park <sangsu@gmail.com>
Date: Fri, 2 Jan 2009 01:11:14 +0900
Subject: USB: Fix goku_udc usb speed handling
Cc: stern@rowland.harvard.edu, david-b@pacbell.net
Message-ID: <20090101161114.GA7977@sangsu-desktop>
Content-Disposition: inline

From: SangSu Park <sangsu@gmail.com>

The usb gadget framework revealed weakness in the godu_udc
gadget driver register function. Instead of checking if
speed asked for was USB_LOW_SPEED upon usb_gadget_register()
to deny service, it checked only for USB_FULL_SPEED, thus
denying service to usb high speed capable gadgets.

Signed-off-by: SangSu Park <sangsu@gmail.com>
Acked-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/gadget/goku_udc.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/gadget/goku_udc.c
+++ b/drivers/usb/gadget/goku_udc.c
@@ -1349,7 +1349,7 @@ int usb_gadget_register_driver(struct us
 	int			retval;
 
 	if (!driver
-			|| driver->speed != USB_SPEED_FULL
+			|| driver->speed < USB_SPEED_FULL
 			|| !driver->bind
 			|| !driver->disconnect
 			|| !driver->setup)
